### ISSUE TITLE
Alpha channel disappears ?

### ISSUE THREAD

**Issue by Antr2005** [strukturag/libheif#495]
Maybe it's a avif-coding error, but just in case:
Test files: https://git.ditoforge.com/mirrors/aom-libavif/src/commit/7637023eabb7589a21358f244ccb8a631c210f10/tests/inputs

heif-convert alpha420.avif alpha420.png (alpha420.png empty)

**Comment by farindk** [comment]
Right. The above change should fix it.

**Comment by Antr2005** [comment]
Thank you, maybe there's a small problem, according to https://git.ditoforge.com/mirrors/aom-libavif/commit/7637023eabb7589a21358f244ccb8a631c210f10, alpha420.avif created with `colorist generate "32x32,#ff000080" --yuv 420 inputs/alpha420.avif`, I think, alpha=128 (0x80), after heif_image_get_plane_readonly (heif_channel_interleaved), alpha=150.
Is that how it should be?

**Comment by farindk** [comment]
Fixed. The reason was that the alpha image was stored as a YCbCr image with both color channels filled with 0. That resulted in a YCbCr->RGB conversion outside the RGB color-space and when converting back, alpha was changed from 128 to 150. It is more common to fill the color-planes with 128, which does not give this problem.

However, the whole color-conversion in the alpha-plane is unnecessary and I have disabled it.
